src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllersAsServices.cs

```diff
@@ -26,7 +26,14 @@ public static void AddControllersAsServices(IServiceCollection services, IEnumer
                 throw new ArgumentNullException(nameof(types));
             }
 
-            var controllerTypeProvider = new StaticControllerTypeProvider();
+            StaticControllerTypeProvider controllerTypeProvider = null;
+
+            var existingServiceDescriptor = services.SingleOrDefault(s => s.ServiceType == typeof(IControllerTypeProvider));
```

Comment:
`FirstOrDefault`? We don't actually care if there's more than one `IControllerTypeProvider`, do we?